From b2104d46b83ca54a20d73c643011812fa864dfcd Mon Sep 17 00:00:00 2001
From: Ruslan Bay <ruslanbey@proton.me>
Date: Thu, 26 Feb 2026 17:21:17 +0100
Subject: [PATCH 22/56] media: intel/ipu4: Use dma_map_sgtable helpers for
 scatter-gather mapping

Replace deprecated dma_map_sg() and dma_sync_sg_for_device() calls with the
modern dma_map_sgtable() and dma_sync_sgtable_for_device() API. This simplifies
DMA scatter-gather table handling and uses the preferred kernel API.

Replace the deprecated dma_map_sg() and dma_sync_sg_for_device()
calls with dma_map_sgtable() and dma_sync_sgtable_for_device().

Link: https://github.com/intel/ipu6-drivers/blob/master/drivers/media/pci/intel/ipu-buttress.c#L778
---
 drivers/media/pci/intel/ipu-buttress.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/media/pci/intel/ipu-buttress.c b/drivers/media/pci/intel/ipu-buttress.c
index 7967fe5410d3..b6c5e590d052 100644
--- a/drivers/media/pci/intel/ipu-buttress.c
+++ b/drivers/media/pci/intel/ipu-buttress.c
@@ -778,8 +778,8 @@ int ipu_buttress_map_fw_image(struct ipu_bus_device *sys,
 {
 	struct page **pages;
 	const void *addr;
-	unsigned long n_pages, i;
-	int rval;
+	unsigned long n_pages;
+	int rval, i;
 
 	n_pages = PAGE_ALIGN(fw->size) >> PAGE_SHIFT;
 
@@ -806,14 +806,14 @@ int ipu_buttress_map_fw_image(struct ipu_bus_device *sys,
 		goto out;
 	}
 
-	n_pages = dma_map_sg(&sys->dev, sgt->sgl, sgt->nents, DMA_TO_DEVICE);
-	if (n_pages != sgt->nents) {
+	rval = dma_map_sgtable(&sys->dev, sgt, DMA_TO_DEVICE, 0);
+	if (rval < 0) {
 		rval = -ENOMEM;
 		sg_free_table(sgt);
 		goto out;
 	}
 
-	dma_sync_sg_for_device(&sys->dev, sgt->sgl, sgt->nents, DMA_TO_DEVICE);
+	dma_sync_sgtable_for_device(&sys->dev, sgt, DMA_TO_DEVICE);
 
 out:
 	kfree(pages);
-- 
2.51.0

